File: api/src/endpoint/migration/list_github_installation_repositories.rs
use crate::{endpoint::Endpoint, resource::migration::GitHubRepositoryResource};

pub struct ListGitHubInstallationRepositories;

impl Endpoint for ListGitHubInstallationRepositories {
    const PATH: &'static str = "/migration/github/{installation_id}/repositories";
    const METHOD: http::Method = http::Method::GET;

    type Request = ();
    type Response = ListGitHubInstallationRepositoriesResponse;
}

pub type ListGitHubInstallationRepositoriesResponse = Vec<GitHubRepositoryResource>;

File: gitdot-api-ts/src/endpoint/migration/github/list-github-installation-repositories.ts
import { z } from "zod";

import { GitHubRepositoryResource } from "../../../resource";

export const ListGitHubInstallationRepositoriesRequest = z.object({});
export type ListGitHubInstallationRepositoriesRequest = z.infer<
  typeof ListGitHubInstallationRepositoriesRequest
>;

export const ListGitHubInstallationRepositories = {
  path: "/migration/github/{installation_id}/repositories",
  method: "GET",
  request: ListGitHubInstallationRepositoriesRequest,
  response: z.array(GitHubRepositoryResource),
} as const;
export type ListGitHubInstallationRepositories =
  typeof ListGitHubInstallationRepositories;
